File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        strsDict = dict()
        for str0 in strs:
            sortedStr0 = "".join(sorted(str0))
            if sortedStr0 in strsDict:
                strsDict[sortedStr0].append(str0)
            else:
                strsDict[sortedStr0] = [str0]
        # res = []
        # for key, str0 in strsDict.items():
        #     res.append(strsDict[key])
        # return res
        return list(strsDict.values())

File: 49-group-anagrams/group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const map = new Map<string, string[]>();
    for (let i = 0; i < strs.length; i++) {
        const sortedStr = strs[i].split("").sort().join("");
        if (map.has(sortedStr)) {
            map.get(sortedStr).push(strs[i]);
        } else {
            map.set(sortedStr, [strs[i]]);
        }
    }
    return Array.from(map.values());
};